### PR TITLE
[Website] Add release note for 0.17.1

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: _release/0.17.1.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+# Apache Arrow 0.17.1 (18 May 2020)
+
+This is a major release covering more than 1 months of development.
```
Comment (nealrichardson) [inline _release/0.17.1.md:27]:
```suggestion
This is a patch release.
```

Comment (nealrichardson) [reply _release/0.17.1.md:27]: Maybe also add a note why we did a patch release, what components were affected? I think that's been done in the past.

File: _release/0.17.1.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+     2	Antoine Pitrou
+     1	Eric Erhardt
+     1	Joris Van den Bossche
+     1	Micah Kornfield
+     1	Paul Taylor
+     1	Yibo Cai
+     1	Yuan Zhou
+```
+
+## Patch Committers
```
Comment (nealrichardson) [inline _release/0.17.1.md:58]: Delete this?

Comment (kszucs) [reply _release/0.17.1.md]: Bug in changelog.py, already created an issue for it.

File: _release/0.17.1.md (hunk 1 above)
Comment (kszucs) [reply _release/0.17.1.md]: Updated.

**Pull request merged**